crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -541,34 +572,49 @@ impl BaseClient {
     /// otherwise. https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md#room-list-parameters
     fn process_sliding_sync_room_membership(
         &self,
-        room_data: &http::response::Room,
         state_events: &[AnySyncStateEvent],
+        invite_state: &Option<Vec<(Raw<AnyStrippedStateEvent>, AnyStrippedStateEvent)>>,
```

Comment:
nit: in Rust it's more idiomatic to pass `Option<&T>` rather than `&Option<T>`, by calling `as_ref()` at the caller's call-site.

Alternatively, you could make this non `Option`al and have the caller `.unwrap_or_default()`.

Reply:
Could we name this `stripped_state`, though, since it applies to both invites and knocked rooms?